### PR TITLE
Systems lua to json

### PR DESCRIPTION
**Pull request by impaktor**
Convert the local stars and bright stars tables to json (from lua)

~Commit message:~

Note following comments could not be carried over:

- ~44 Boötis B/C are a W-Ursae-Majoris-type shallow contact binary system,
  both G class,~

- ~'40 Eridani' - Problem: the secondary and tertiary are co-orbital,
   and as a system orbit the primary.~

- ~'Mu Herculis' - Problem: the secondary and tertiary are co-orbital,
   and as a system orbit the primary.~

- ~'Alpha Centauri' - Comment: "Gliese 559", "FK5 538", "GC 19728",
   "CCDM J14396-6050"~

__EDIT:__ I've updated above comments into the `comment` field I did not know our json format had.

### CODE REVIEW

**Comment by impaktor** [comment]
Comparison:

# Master
master-sector-2024-08-30-223105_1276x697_scrot

# This branch
branch-sector-2024-08-30-222505_1275x715_scrot

# Master
master-44-2024-08-30-223116_1270x715_scrot

# This branch
branch-44-2024-08-30-222753_1275x721_scrot

# Master
master-system-2024-08-30-223126_1283x722_scrot

# This branch
branch-system-2024-08-30-222806_1274x715_scrot

**Comment by impaktor** [comment]
OK, positions above were off by factor 8, below is script to fix it, and gif now comparing. There's one star that seems to be moving south east of Sol:

## Code
```python
import json

# fpath = "partial/02_local_stars.json"
fpath = "partial/03_bright_stars.json"


with open(fpath) as f:
    systems = json.load(f)

# All Lua values for in-sector position need to be multiplied by 8 to
# turn them into lightyears (TLDR: json uses ly for units, lua uses
# percent-of-a-sector)
for i in range(len(systems)):
    systems[i]['pos'] = [s * 8 for s in systems[i]['pos']]

# Dump as is
# with open('data.json', 'w') as f:
#     json.dump(systems, f)

# Don't json format
out = json.dumps(systems, indent=None)

# Make one system per line:
out = out.replace("}, {", "},\n{")

with open('data.json', 'w') as f:
    f.write(out)
```

## Comparison
sector

**Comment by impaktor** [comment]
Just checked on master. The star that's popping in/out above is "GJ 1075", this was the only star in 02_local_stars.json on master, and was copied over from it's lua counterpart and had already been multiplied with x8, thus I should rollback the x8 my parsing just did for that line.

File: data/systems/partial/02_local_stars.json [hunk 1]
```diff
@@ -1,3 +1,409 @@
...
+    {"name": "Gliese 806", "stars": ["STAR_M"], "sector": [-3, -3, 3], "pos": [2.144, 4.824, 4.568]},
+    {"name": "33 G. Lib", "stars": ["STAR_K"], "sector": [-2, 1, -1], "pos": [3.464, 4.824, 0.968]},
+    {"name": "Gliese 412", "stars": ["STAR_M", "STAR_M"], "sector": [0, 1, 1], "pos": [2.688, 3.104, 2.856]},
+    {"name": "Gliese 361", "stars": ["STAR_M"], "sector": [2, 3, 1], "pos": [4.528, 5.64, 0.464]},
+    {"name": "Epsilon Indi", "stars": ["STAR_K"], "sector": [-1, -1, -2], "pos": [4.84, 2.344, 6.104]},
+    {"name": "Gliese 203", "stars": ["STAR_M"], "sector": [3, -1, 0], "pos": [3.68, 4.112, 4.752]},
+    {"name": "Xi Bo\u00f6tis", "stars": ["STAR_G", "STAR_K"], "sector": [-2, 1, 0], "pos": [1.96, 7.136, 7.152]},
+    {"name": "Gliese 205", "stars": ["STAR_M"], "sector": [2, -1, -1], "pos": [2.376, 5.696, 6.808]},
+    {"name": "NN 3976", "stars": ["STAR_M"], "sector": [-4, 1, 1], "pos": [3.216, 0.944, 4.456]},
+    {"name": "Alpha Centauri", "stars": ["STAR_G", "STAR_K"], "otherNames": ["Bungula", "Toliman", "\u03b1 Centauri", "Rigil Kentaurus", "Gliese 551", "Gliese 559", "FK5 538", "HIP 70890"], "sector": [-1, 0, -1], "pos": [6.624, 1.632, 4.16]},
```
Comment (Web-eWorks) [inline data/systems/partial/02_local_stars.json:224]: These should be using utf-8 codepoints, not ASCII-fied UTF-16 escapes (which is the default python `json.dump` behavior).

Comment (impaktor) [reply data/systems/partial/02_local_stars.json]: fixed

File: data/systems/partial/02_local_stars.json [hunk 2]
```diff
@@ -1,3 +1,409 @@
...
+    {"name": "Sigma Draconis", "stars": ["STAR_K"], "sector": [-1, -1, 2], "pos": [1.984, 5.432, 1.64]},
+    {"name": "GJ 2034", "stars": ["WHITE_DWARF"], "sector": [1, -1, -3], "pos": [7.912, 0.656, 4.096]},
+    {"name": "GJ 2036", "stars": ["STAR_M", "STAR_M"], "sector": [2, -1, -4], "pos": [3.608, 2.152, 1.824]},
+    {"name": "Gliese 109", "stars": ["STAR_M"], "sector": [1, -3, 1], "pos": [6.6, 7.24, 2.616]},
+    {"name": "Alpha Fornacis", "stars": ["STAR_F", "STAR_G"], "sector": [3, -4, -3], "pos": [5.928, 5.072, 1.696]},
+    {"name": "Pollux", "stars": ["STAR_K_GIANT"], "sector": [3, 1, 1], "pos": [2.672, 5.2, 7.84]},
+    {"name": "Gliese 102", "stars": ["STAR_M"], "sector": [2, -3, 1], "pos": [1.944, 1.368, 5.432]},
+    {"name": "Gliese 103", "stars": ["STAR_K"], "sector": [2, -3, -4], "pos": [0.904, 2.816, 6.016]},
+    {"name": "Upsilon Andromedae", "stars": ["STAR_F"], "sector": [1, -4, 3], "pos": [5.504, 1.944, 5.056]},
+    {"name": "Luyten\"s Star", "stars": ["STAR_M"], "sector": [1, 0, 0], "pos": [3.448, 4.592, 1.128]},
```
Comment (Web-eWorks) [inline data/systems/partial/02_local_stars.json:89]: One final issue - the apostrophe-to-double quote path is converting this Lua-side escaped apostrophe into an escaped double-quote. Feel free to manually fix it (with a grep for other occurrences).

**Review by Web-eWorks — APPROVED** [review]

**Pull request merged**